src/types/reducers.js

```diff
@@ -68,22 +68,27 @@ export type RangeFilterState = {
   end: number,
 };
 
-export type UrlState = {
+export type SearchStringState = {|
+  +current: string,
+  +stack: string[],
+|};
```

Comment:
I started to use the contravariance to flow-force the fact that state should never be written to.